### DISCUSSION TITLE
DIPY Project#1 "Add mutual information for non-rigid registration"

### DISCUSSION THREAD

**Discussion by WL678** [dipy/dipy#2737] [#GSoC/GSoD]
Hello mentors,
I'm Autoni Kevin and I will be doing a Masters at PKU in China next year. I am fascinated with project #1 - Add mutual information for non-rigid registration - as part of GSoC 2023. 
To be honest, I have been paying attention to GSoC since autumn 2022, and I was really excited when I found the project about image registration! I'm currently working on zebrafish brain registration. I tried to replicate the work of Eftychios A. Pnevmatikakis, who previously used piecewise rigid registration to approximate non-rigid registration (https://doi.org/10.1016/j.jneumeth.2017.07.031). Although the NoRMCorre tool gives good results, it's not open source and has a data format limitation. Therefore, I would like to contribute to the development of open source tools for non-rigid registration. 
I would like to continue researching in this area and this is a really great opportunity for me to do so. I believe I can make a significant contribution to this project if given the opportunity.
I look forward to working with you!

@skoudoro @pjsjongsung

**Comment by skoudoro** [comment]
Hi @WL678 ,

Thank you for your interest in this project. You have a really interesting background and I like your passion for image registration.

To get started, we recommend to take a look at the guidelines on how to contribute to DIPY.

The second step is making multiple small enhancement/bugfix/documentation fix/etc to DIPY before applying for the GSoC (mandatory). It helps you get some idea how things would work during the GSoC. The fix does not need to be related to your proposal.

Do not forget that gsoc is FIRST a program to help and bring value to opensource software and Second the project.

Looking forward for your first contribution!

If you have any specific questions concerning the gsoc project. We will be happy to answer them in this thread.
if you do not know what to do for contributing, you should first start to run the test, some examples and read documentation, read all the issues. You will find naturally what to do to help.

an interesting issue to fix is this one : https://github.com/dipy/dipy/issues/2703.  You will need to create a new tutorial.
other options to look at are:
- https://github.com/dipy/dipy/issues/2648
- https://github.com/dipy/dipy/issues/2571
- https://github.com/dipy/dipy/issues/2535
- https://github.com/dipy/dipy/issues/2496
- and much more if you dig !

**Reply by WL678** [reply]
First,I apologize for the delayed response due to the hectic semester. 

During this period, I tried various methods of image registration, including affine registration, symmetric diffeomorphic registration and Streamline-registration. 
To cope with #2703, I tried my best to studied all the tutorials of dipy about registration. But there are many details I don't understand, which hinder me from solving the problem. I used to know so little about streamline registration that I didn't know exactly what William's problem was. In my opinion, william  had used streamlines_registration.py to estimate reasonable deformation fields. However, when he applied the deformation field to the stremlines, the warped streamlines did not align well with the FA MNI template and warped “raw” FA image template. 

Honestly, I think it is a challenge for me mastering streamline-registration quickly in the graduation season.Even so, I want to keep trying to solve this interesting problem, struggling to know where to start.
I would appreciate if you could provide some suggestions or ideas!

@skoudoro

**Comment by smruthi49** [comment]
Hi,
I am a complete beginner but I am very much interested in knowing about mutual information, non rigid registration. I am excited to learn about it. Open source and the topis in general in new for me. Can I still make progress and contribute?
Please do let me know. I'd be looking forward to work on it even more.
Thank you!
